docs/concept/clustering.md

```diff
@@ -0,0 +1,158 @@
+
+### 3.4 Liaison Nodes
+
+Liaison Nodes do not store data but manage the routing of incoming requests to the appropriate Query or Data Nodes. They also provide authentication, TTL, and other security services.
+
+## 4. **Determining Optimal Node Counts**
+
+When creating a BanyanDB cluster, choosing the appropriate number of each node type to configure and connect is crucial. The number of Meta Nodes should always be odd, for instance, “3”. The number of Data Nodes scales based on your storage and query needs. The number of Query Nodes and Liaison Nodes depends on the expected query load and routing complexity.
+
+The BanyanDB architecture allows for efficient clustering, scaling, and high availability, making it a robust choice for time series data management.
+
+## 5. Writes in a Cluster
+
+In BanyanDB, writing data in a cluster is designed to take advantage of the robust capabilities of underlying storage systems, such as Google Compute Persistent Disk or Amazon S3(TBD). These platforms ensure high levels of data durability, making them an optimal choice for storing raw time series data.
+
+Unlike some other systems, BanyanDB does not support application-level replication. Instead, it delegates the task of replication to these underlying storage systems. This approach simplifies the BanyanDB architecture and reduces the complexity of managing replication at the application level.
+
+Data distribution across the cluster is determined based on the `shard_num` setting for a group and the specified `entity` in each resource, be it a stream or measure. The resource’s `name` with its `entity` is the sharding key, guiding data distribution to the appropriate Data Node during write operations.
+
+Liaison Nodes retrieve shard mapping information from Meta Nodes to achieve efficient data routing. This information is used to route data to the appropriate Data Nodes based on the sharding key of the data.
+
+This sharding strategy ensures the write load is evenly distributed across the cluster, enhancing write performance and overall system efficiency. BanyanDB uses a hash algorithm for sharding. The hash function maps the sharding key (resource name and entity) to a node in the cluster. Each shard is assigned to the node returned by the hash function.
+
+Here's a text-based diagram illustrating the data write path in BanyanDB:
+
+```
+User
+ |
+ | API Request (Write)
+ |
+ v
+------------------------------------
+|          Liaison Node             |   <--- Stateless Node, Routes Request
+| (Identifies relevant Data Nodes   |
+|  and dispatches write request)    |
+------------------------------------
+       |                            
+       v                            
+-----------------  -----------------  -----------------
+|  Data Node 1  |  |  Data Node 2  |  |  Data Node 3  |
+|  (Shard 1)    |  |  (Shard 2)    |  |  (Shard 3)    |
+-----------------  -----------------  -----------------
```

Comment:
> Instead, it delegates the task of replication to these underlying storage systems.

I think diagram is not complete, should we add more detail diagram about how to delegates the task of replication to these underlying storage systems?

Reply:
like :
User
 |
 | API Request (Write)
 |
 v
------------------------------------
|          Liaison Node             |   <--- Stateless Node, Routes Request
| (Identifies relevant Data Nodes   |
|  and dispatches write request)    |
------------------------------------
       |                            
       v                            
-----------------  -----------------  -----------------
|  Data Node 1  |  |  Data Node 2  |  |  Data Node 3  |
|  (Shard 1)    |  |  (Shard 2)    |  |  (Shard 3)    |
       |                          |                           |
       v                         v                         v
---------------------------------------------------
                              shared storage

Reply:
Make sense. You could push a "suggestion" to update the text diagram.

Reply:
Sorry. The data node doesn't necessarily have to be built on "shared" storage. It just requires a robust one in case of a potential single point of failure. Therioally, the whole architecture is shared-nothing instead of shared storage.